### ISSUE TITLE
Apply `escape-case` to regex literal sequences

### ISSUE THREAD

**Issue by brettz9** [sindresorhus/eslint-plugin-unicorn#273]
[<img alt="Issuehunt badges" src="https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/273/badge.svg" />](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/273)

Thanks for the excellent set of rules!

I'm wondering if you might allow `escape-case` to be applied to regex literal sequences as well as strings:

## Fail

```js
const foo = /\xa9/;
const foo = /\ud834/;
const foo = /\u{1d306}/u;
const foo = /\ca/;
```

## Pass

```js
const foo = /\xA9/;
const foo = /\uD834/;
const foo = /\u{1D306}/u;
const foo = /\cA/;
```

(Incidentally, while these examples I adopted are suitable for my case, are you sure you want to be supporting them in strings, given that `\c` does not relate to control characters in strings?)

> There is a $60.00 open bounty on this issue. Add more on [Issuehunt](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/issues/273).
> - Checkout the [Issuehunt explorer](https://issuehunt.io/r/sindresorhus/eslint-plugin-unicorn/) to discover more funded issues.
> - Need some help from other developers? [Add your repositories](https://issuehunt.io/r/new) on Issuehunt to raise funds.

**Comment by sindresorhus** [comment]
Yeah, I think this would be useful. We could use https://github.com/mysticatea/regexpp to parse the regex.

@futpib Thoughts?

---

> are you sure you want to be supporting them in strings, given that \c does not relate to control characters in strings?)

Good catch. We should not. That seems like a bug.

Reference: https://mathiasbynens.be/notes/javascript-escapes

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $60.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/55832243/issues/273) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/55832243/issues/273).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/r/new) on IssueHunt to raise funds.

**Comment by IssueHuntBot** [comment]
@sindresorhus has rewarded $54.00 to @jlhwung. [See it on IssueHunt](https://issuehunt.io/repos/55832243/issues/273)

- :moneybag: Total deposit: $60.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $6.00
